test/gui/shared/scripts/pageObjects/PublicLinkDialog.py

```diff
@@ -168,22 +168,50 @@ def setExpirationDate(self, expireDate):
                 squish.Qt.NoModifier,
                 squish.Qt.LeftButton,
             )
-            squish.nativeType("<Delete>")
-            squish.nativeType("<Delete>")
-            squish.nativeType(expDate.month)
-            squish.nativeType(expDate.day)
+            # Move the cursor to year (last) field and enter the year
+            squish.nativeType("<Ctrl+Right>")
+            squish.nativeType("<Ctrl+Right>")
             squish.nativeType(expYear)
+
+            # Move the cursor to day (middle) field and enter the day
+            squish.nativeType("<Ctrl+Left>")
+            squish.nativeType(expDate.day)
+
+            # Move the cursor to month (first) field and enter the month
+            # Backspace works most of the time, so we clear the data using backspace
+            squish.nativeType("<Ctrl+Left>")
+            squish.nativeType("<Ctrl+Left>")
+            squish.nativeType("<Right>")
+            squish.nativeType("<Backspace>")
+            squish.nativeType("<Backspace>")
+            squish.nativeType(expDate.month)
             squish.nativeType("<Return>")
 
             actualDate = str(
                 squish.waitForObjectExists(self.EXPIRATION_DATE_FIELD).displayText
             )
             expectedDate = f"{expDate.month}/{expDate.day}/{expYear}"
+
             if not actualDate == expectedDate:
+                test.log(
+                    f"Expected date: {expectedDate} is not same as that of Actual date: {actualDate}, trying workaround"
+                )
                 # retry with workaround
                 self.setExpirationDateWithWorkaround(
                     expYear, expDate.month, expDate.day
                 )
+                if not actualDate == expectedDate:
+                    test.warning(
+                        f"workaround failed, actual date: {actualDate} is still not same as expected date: {expectedDate}"
+                    )
```

Comment:
`actualDate` will not get updated here so you have to recalculate again. Also maybe throw from here if the dates are not equal (the test is going to fail anyway).